pkg/commands/plan/plan.go

```diff
@@ -4,61 +4,281 @@
+		if err := c.githubClient.UpdateIssueComment(
+			ctx,
+			c.cfg.RepositoryOwner,
+			c.cfg.RepositoryName,
+			startComment.GetID(),
+			fmt.Sprintf("%s 🟦 No Terraform files have changes, planning skipped.", planCommentPrefix),
+		); err != nil {
+			merr = errors.Join(merr, fmt.Errorf("failed to update plan comment: %w", err))
+		}
+	} else {
+		if err := c.githubClient.DeleteIssueComment(
+			ctx,
+			c.cfg.RepositoryOwner,
+			c.cfg.RepositoryName,
+			startComment.GetID(),
+		); err != nil {
+			merr = errors.Join(merr, fmt.Errorf("failed to delete plan comment: %w", err))
+		}
+	}
+
+	c.Outf("\n\nGuardian Plan completed")
 
-  Run the Terraform plan for a directory.
-`
+	return merr
 }
 
-func (p *PlanCommand) Flags() *cli.FlagSet {
-	set := cli.NewFlagSet(p.testFlagSetOpts...)
-	return set
+// handleTerraformPlan runs the required Terraform commands for a full run of
+// a Guardian plan using the Terraform CLI.
+func (c *PlanCommand) handleTerraformPlan(ctx context.Context) (*Result, error) {
+	var stdout, stderr strings.Builder
+	multiStdout := io.MultiWriter(c.Stdout(), &stdout)
+	multiStderr := io.MultiWriter(c.Stderr(), &stderr)
+
+	lockfileMode := "none"
+	if c.flagProtectLockfile {
+		lockfileMode = "readonly"
+	}
+
+	c.actions.Group("Initializing Terraform")
+
+	if _, err := c.terraformClient.Init(
+		ctx,
+		c.Stdout(),
+		multiStderr,
+		&terraform.InitOptions{
+			Input:       util.Ptr[bool](false),
+			NoColor:     util.Ptr[bool](true),
+			Lockfile:    util.Ptr[string](lockfileMode),
+			LockTimeout: util.Ptr[string](c.flagLockTimeout.String()),
+		}); err != nil {
+		return &Result{commentDetails: stderr.String()}, fmt.Errorf("failed to initialize: %w", err)
+	}
+	stderr.Reset()
+	c.actions.EndGroup()
+
+	c.actions.Group("Validating Terraform")
+
+	if _, err := c.terraformClient.Validate(
+		ctx,
+		c.Stdout(),
+		multiStderr,
+		&terraform.ValidateOptions{NoColor: util.Ptr[bool](true)},
+	); err != nil {
+		return &Result{commentDetails: stderr.String()}, fmt.Errorf("failed to validate: %w", err)
+	}
+
+	stderr.Reset()
+	c.actions.EndGroup()
+
+	c.actions.Group("Running Terraform plan")
+
+	planExitCode, err := c.terraformClient.Plan(ctx, c.Stdout(), multiStderr, &terraform.PlanOptions{
+		Out:              util.Ptr[string](c.planFilename),
+		Input:            util.Ptr[bool](false),
+		NoColor:          util.Ptr[bool](true),
+		DetailedExitcode: util.Ptr[bool](true),
+		LockTimeout:      util.Ptr[string](c.flagLockTimeout.String()),
+	})
+
+	stderr.Reset()
+	c.actions.EndGroup()
+
+	hasChanges := planExitCode == 2
+
+	if err != nil && !hasChanges {
+		return &Result{commentDetails: stderr.String()}, fmt.Errorf("failed to plan: %w", err)
+	}
+
+	if !hasChanges {
+		return &Result{hasChanges: hasChanges}, nil
+	}
+
+	c.actions.Group("\n\nFormatting plan output")
+
+	_, err = c.terraformClient.Show(ctx, multiStdout, multiStderr, &terraform.ShowOptions{File: util.Ptr[string](c.planFilename), NoColor: util.Ptr[bool](true)})
```

Comment:
Do you really need to box these generics? `util.Ptr` should know what it's giving/getting...

Reply:
Wasnt sure, I will remove.